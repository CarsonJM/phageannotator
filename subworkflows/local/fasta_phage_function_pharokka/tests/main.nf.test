nextflow_workflow {

    name "Test Subworkflow: FASTA_PHAGE_FUNCTION_PHAROKKA"
    script "../main.nf"
    workflow "FASTA_PHAGE_FUNCTION_PHAROKKA"
    config "./nextflow.config"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "fasta_phage_function_pharokka"
    tag "fasta_phage_function_pharokka_default"


    test("fasta") {

        when {
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                    ]
                )
                input[1] = null
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
