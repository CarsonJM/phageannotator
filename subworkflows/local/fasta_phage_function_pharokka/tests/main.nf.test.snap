{
    "Parameters: default": {
        "content": [
            {
                "0": [
                    "versions.yml:md5,7715e330cd5621f95141e256b567ec19",
                    "versions.yml:md5,b225eb2829469045a5bda2c5d7a69009"
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gbk.gz:md5,e719cd124ccca0524df508cd178e853e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test_cds_final_merged_output.tsv:md5,1242b6ffe6e53ef5bce0a9a2cac3cd45"
                    ]
                ],
                "pharokka_final_output_tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_cds_final_merged_output.tsv:md5,1242b6ffe6e53ef5bce0a9a2cac3cd45"
                    ]
                ],
                "pharokka_gbk_gz": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gbk.gz:md5,e719cd124ccca0524df508cd178e853e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7715e330cd5621f95141e256b567ec19",
                    "versions.yml:md5,b225eb2829469045a5bda2c5d7a69009"
                ]
            }
        ],
        "timestamp": "2024-01-24T14:00:39.927815714"
    }
}
